text/0000-pack-package-refactor.md

```diff
@@ -0,0 +1,118 @@
+# Meta
+
+- Name: Refactor Pack go packages
+- Start Date: 02/24/21
+- Author(s): dwillist
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+This RFC outlines a promotion strategy and process for refactoring `pack` into consumable components as needed.
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?
+    - Establish some shared vision about the codebase we want to manage.
+    - General refactoring benifits: 
+        - Maintaining a modular and re-usable codebase enables contributors to be more effective, and encourages new adopters.
+        - Ease consumption of pack as a library by having isolated and focused packages
+        - Create logical seams to redistribute packages.
+
+# What it is
+[what-it-is]: #what-it-is
+The general flow of this process is
+1) Unorganized features exist & are added to the `internal` package, these are not consumable by external programs
+2) Extract code from the `interal` package to `/pkg/<pkg-name>/...` to enable external consumption.
```

Comment:
```suggestion
1) Extract code from the `interal` package to `/pkg/<pkg-name>/...` to enable external consumption.
```
etc all the way down, to ease in case of reordering